allennlp/semparse/contexts/table_question_context.py

```diff
@@ -81,17 +82,27 @@ class TableQuestionContext:
     for extracting entities from a question given a table and type its columns with <string> | <date> | <number>
     """
     def __init__(self,
-                 cell_values: Set[str],
-                 column_type_statistics: List[Dict[str, int]],
-                 column_index_to_name: Dict[int, str],
+                 table_data: List[Dict[str, str]],
+                 column_types: Dict[str, str],
                  question_tokens: List[Token]) -> None:
-        self.cell_values = cell_values
-        self.column_types = {column_index_to_name[column_index]: max(column_type_statistics[column_index],
-                                                                     key=column_type_statistics[column_index].get)
-                             for column_index in column_index_to_name}
+        self.table_data = table_data
```

Comment:
You're taking an input here, then immediately overriding it below.  Something is off.

Reply:
Oh, thanks! Fixed it and added some comments.